File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {

		Map<Integer, Integer> count = new HashMap<>();

		PriorityQueue<Map.Entry<Integer, Integer>> maxHeap = new PriorityQueue<>((a, b) -> b.getValue() - a.getValue());

		for (int num: nums){
			count.put(num, count.getOrDefault(num,0)+1);
		}

		maxHeap.addAll(count.entrySet());

		int[] result = new int[k];
		for (int i = 0; i < k; i++) {
			result[i] = maxHeap.poll().getKey();
		}

		return result;



        
    }
}

File: Data Structures & Algorithms/subarray-sums-divisible-by-k/submission-0.py
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        mp = defaultdict(int)
        result = 0
        prefix = 0
        mp[0] = 1

        for num in nums:
            prefix += num

            remainder = prefix % k

            result += mp[remainder]
            mp[remainder]+= 1
        
        return result